packages/sync/src/conflicts/VersionConflictHandler.ts

```diff
@@ -2,39 +2,41 @@ import { ConflictResolutionData } from "./strategies/ConflictResolutionData";
 import { ConflictListener } from "./strategies/ConflictListener";
 import { ConflictResolutionStrategy } from "./strategies/ConflictResolutionStrategy";
 import { ConflictHandler, HandlerOptions } from "./ConflictHandler";
+import { ObjectState } from "./state/ObjectState";
 
-export class ConflictHandlerVersioned implements ConflictHandler {
+export class VersionConflictHandler implements ConflictHandler {
   public conflicted: boolean;
   public clientDiff: any = {};
   public serverDiff: any = {};
   public options: HandlerOptions;
+  private ignoredKeys: string[];
 
   constructor(options: HandlerOptions) {
     this.options = options;
     this.conflicted = false;
+    this.ignoredKeys = this.options.ignoredKeys || ["version", "id"];
```

Comment:
This is the only place we put version. Why we really want to have this as versioned. The entire implementation is generic and this makes things much more complex for the end dev.